Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -460,26 +466,35 @@ def is_oms_primary_workspace(workspace_id):
     else:
         return False
 
-
-def get_workspaceid_agentid_from_oms_config():
-    # Reads the oms config file
-    # Returns: AgentID config value
+def read_omsconfig_file():
     if os.path.isfile(OMS_ADMIN_CONFIG_FILE):
         # the above path always points to the oms configuration file of the primary workspace
-        try:
-            keyvals = config_file_to_kv_pair(OMS_ADMIN_CONFIG_FILE)
-            return keyvals[OPTION_OMS_WORKSPACE_ID].strip(), keyvals[OPTION_AGENT_ID].strip()
-        except ConfigParser.NoSectionError, exception:
-            log(DEBUG, str(exception))
-            raise ConfigParser.Error(str(exception))
-        except ConfigParser.NoOptionError, exception:
-            log(DEBUG, str(exception))
-            raise ConfigParser.Error(str(exception))
+        keyvals = config_file_to_kv_pair(OMS_ADMIN_CONFIG_FILE)
+        return keyvals
     else:
         error_string = "could not find file " + OMS_ADMIN_CONFIG_FILE
         log(DEBUG, error_string)
         raise ConfigParser.Error(error_string)
 
+def get_azure_resource_id_from_oms_config():
+    keyvals = read_omsconfig_file()
+    try:
+        return keyvals[OPTION_AZURE_RESOURCE_ID].strip()
+    except KeyError:
+        log(DEBUG, "Azure resource id was not spcified in omsadming config file")
+        return ""
+
+def get_workspaceid_agentid_from_oms_config():
+    # Reads the oms config file
+    # Returns: AgentID config value
+
+    keyvals = read_omsconfig_file()
+    try:
+        return keyvals[OPTION_OMS_WORKSPACE_ID].strip(), keyvals[OPTION_AGENT_ID].strip()
+    except KeyError, exception:
+        log(DEBUG, str(exception))
```

Comment:
The exception is caught and thrown again where it will trigger an Error log level. Lower log levels are for helper functions that I use. I try to have higher log levels in the main workflow only so that we have fewer duplicates of higher log levels.